sale_purchase_customer/views/purchase_order.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="purchase_order_form_view" model="ir.ui.view">
+        <field name="name">purchase.order.form.view</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_ref']" position="after">
+                <field name="sale_partner_id" attrs="{'readonly':[('state','in',['done','cancel'])]}"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="purchase_order_search_view" model="ir.ui.view">
+        <field name="name">purchase.order.search.view</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.view_purchase_order_filter" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="after">
+                <field name="sale_partner_id" />
+            </xpath>
+            <xpath expr="//filter[@name='order_date']" position="after">
```

Comment:
This xpath matches with two nodes at the original search view. Please ensure that we put it within `<group>` search view section (the Group By section indeed), because this is the effect (an unexpected separator, and actually _is not before Order date_, is it?):

![image](https://github.com/user-attachments/assets/b7d07eeb-6274-4439-a24c-87aff20c2a8b)